superbench/analyzer/data_diagnosis.py

```diff
@@ -406,12 +407,11 @@ def run(
                 output_path = str(Path(output_dir) / 'diagnosis_summary.xlsx')
                 self.output_diagnosis_in_excel(self._raw_data_df, data_not_accept_df, output_path, self._sb_rules)
             elif output_format == 'json':
-                if output_all:
-                    output_path = str(Path(output_dir) / 'diagnosis_summary.json')
-                    self.output_diagnosis_in_json(data_not_accept_df, output_path)
-                else:
-                    output_path = str(Path(output_dir) / 'diagnosis_summary.jsonl')
-                    self.output_diagnosis_in_jsonl(data_not_accept_df, output_path)
+                output_path = str(Path(output_dir) / 'diagnosis_summary.json')
+                self.output_diagnosis_in_json(data_not_accept_df, output_path)
+            elif output_format == 'jsonl':
+                output_path = str(Path(output_dir) / 'diagnosis_summary.jsonl')
+                self.output_diagnosis_in_jsonl(data_not_accept_df, output_path)
```

Comment:
do you support --output-all in jsonl? why output path is set to json if output_all
https://github.com/microsoft/superbenchmark/blob/c0e7200775ba88f8f5c43c9633bb7f0d19a0e89f/superbench/analyzer/data_diagnosis.py#L402-L403

Reply:
good catch, this line is useless, I will remove it